source/fuzz/transformation_add_bit_instruction_synonym.cpp

```diff
@@ -111,6 +116,48 @@ void TransformationAddBitInstructionSynonym::Apply(
   }
 }
 
+bool TransformationAddBitInstructionSynonym::IsInstructionOperandsHasSameSign(
+    opt::IRContext* ir_context, opt::Instruction* instruction) {
+  if (instruction->opcode() == SpvOpNot) {
+    auto operand = instruction->GetInOperand(0).words[0];
+    auto operand_inst = ir_context->get_def_use_mgr()->GetDef(operand);
+    auto operand_type =
+        ir_context->get_type_mgr()->GetType(operand_inst->type_id());
+    auto operand_sign = operand_type->AsInteger()->IsSigned();
+
+    auto type_id_sign = ir_context->get_type_mgr()
+                            ->GetType(instruction->type_id())
+                            ->AsInteger()
+                            ->IsSigned();
+
+    return operand_sign != type_id_sign;
```

Comment:
This should be `operand_sign == type_id_sign` since the function is supposed to return `true` when the operands have the same signedness.

Reply:
Done.

Reply:
If the function returns `true` you want to return `false` from `IsApplicable`?